pkg/operator/mon/mon.go

```diff
@@ -205,13 +197,14 @@ func (c *Cluster) failoverMon(conn client.Connection, name string) error {
 	logger.Infof("Failing over monitor %s", name)
 
 	// Start a new monitor
-	c.maxMonID++
-	mons := []*MonConfig{&MonConfig{Name: fmt.Sprintf("mon%d", c.maxMonID), Port: int32(mon.Port)}}
+	mons := []*MonConfig{&MonConfig{Name: fmt.Sprintf("mon%d", c.maxMonID+1), Port: int32(mon.Port)}}
```

Comment:
maxMonID is the largest existing mon ID, so to create another mon we have to increment.